gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoRestSerializer.java

```diff
@@ -159,4 +161,19 @@ public String toBody(String fieldName, RequestT fieldValue) {
   public String toBody(String fieldName, RequestT fieldValue, boolean numericEnum) {
     return toJson(fieldValue, numericEnum);
   }
+
+  /**
+   * Serializes an object to a query parameter Handles the case of a message such as Duration,
+   * FieldMask or Int32Value to prevent wrong formatting that String.valueOf() would make
+   *
+   * @param fieldValue a field value to serialize
+   */
+  public String toQueryParamValue(Object fieldValue) {
+    if (fieldValue instanceof GeneratedMessageV3) {
```

Comment:
I'm not sure we want to handle this for all GeneratedMessageV3, this basically means we want to handle any complex types.

Reply:
@vam-google I saw that Noah came up with [a list of wellknownTypes](https://github.com/googleapis/gapic-showcase/blob/fe414784c18878d704b884348d84c68fd6b87466/util/genrest/resttools/populatefield.go#L27), do you think this is a full list for supported types in query params?

Reply:
I added the well known types to the logic. The check for GeneratedMessageV3 in `toQueryParamValue` will be intersected with serializable message types (e.g. FieldMask) with an extra condition in this if statement - also added a comment line for clarification)